Scripts/UI/GradientButton.cs

```diff
@@ -92,23 +110,57 @@ public bool visible
 
 				m_Visible = value;
 
+				if (m_Visible && !gameObject.activeSelf)
+					gameObject.SetActive(true);
+
 				this.StopCoroutine(ref m_VisibilityCoroutine);
 				m_VisibilityCoroutine = value ? StartCoroutine(AnimateShow()) : StartCoroutine(AnimateHide());
 			}
 		}
 		bool m_Visible;
 
+		Vector3 originalScale
+		{
+			get { return invertHighlightScale ? m_HighlightContentContainerLocalScale : m_OriginalContentContainerLocalScale; }
+		}
+
+		Vector3 highlightScale
+		{
+			get { return invertHighlightScale ? m_OriginalContentContainerLocalScale : m_HighlightContentContainerLocalScale; }
+		}
+
+		public Sprite icon
```

Comment:
This is not used, and seems to conflict with iconSprite above.

In fact, a lot of these public properties are not used or are only used privately.  Since it's an internal class, I would say that we should get rid of any of the stuff we're not using, or document it and make this a public class.